internal/pkg/merge/merge.go

```diff
@@ -0,0 +1,137 @@
+// ------------------------------------------------------------------------
+// SPDX-FileCopyrightText: Copyright © 2024 bomctl a Series of LF Projects, LLC
+// SPDX-FileName: internal/pkg/merge/merge.go
+// SPDX-FileType: SOURCE
+// SPDX-License-Identifier: Apache-2.0
+// ------------------------------------------------------------------------
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ------------------------------------------------------------------------
+package merge
+
+import (
+	"fmt"
+	"slices"
+
+	"github.com/google/uuid"
+	"github.com/protobom/protobom/pkg/sbom"
+	"google.golang.org/protobuf/types/known/timestamppb"
+
+	"github.com/bomctl/bomctl/internal/pkg/db"
+	"github.com/bomctl/bomctl/internal/pkg/options"
+)
+
+func Merge(documentIDs []string, opts *options.MergeOptions) (string, error) {
+	backend, err := db.BackendFromContext(opts.Context())
+	if err != nil {
+		return "", fmt.Errorf("%w", err)
+	}
+
+	backend.Logger.Info("Merging documents", "documentIDs", documentIDs)
+
+	documents, err := backend.GetDocumentsByID(documentIDs...)
+	if err != nil {
+		return "", fmt.Errorf("%w", err)
+	}
+
+	// Make document list a map so it can sort by the ids proved
+	documentMap := make(map[string]*sbom.Document)
+	for _, doc := range documents {
+		documentMap[doc.Metadata.Id] = doc
+	}
+
+	merged, err := performTopLevelMerge(documentIDs, documentMap, opts)
+	if err != nil {
+		backend.Logger.Fatal(err)
```

Comment:
should these not return and terminate the function?

Reply:
The `Fatal` method is a terminating call

Reply:
@pkwiatkowski1 Although any time you can return an error it's better to do that than panic outright. Gives the calling context a chance to handle it or wrap the error

Reply:
I have it updated to return on all of these, missed adding the returns for the new methods.